lkmpg.tex

```diff
@@ -219,17 +219,44 @@ \subsection{The Simplest Module}
 \begin{code}
 obj-m += hello-1.o
 
+PWD := $(CURDIR)
+
 all:
 	make -C /lib/modules/$(shell uname -r)/build M=$(PWD) modules
 
 clean:
 	make -C /lib/modules/$(shell uname -r)/build M=$(PWD) clean
 \end{code}
 
-And finally just:
+And finally just run \verb|make| directly.
 \begin{codebash}
 make
 \end{codebash}
+If there is no \verb|PWD := $(CURDIR)| statement in Makefile, note that using \verb|sudo make| might not compile. Because the \verb|sudo| will fork the process. According to the Process Model section in \href{https://man7.org/linux/man-pages/man8/sudo.8.html}{sudo manual}, some of the environment variables won't be inherited. Here is the \verb|PWD| variable which will be passed to the kernel Makefile. For example, here is the simple Makefile:
```

Comment:
Change "might not compile." into "might not compile correctly.".
Also, it would be better to briefly explain the effect of $(CURDIR):
```
Set to the absolute pathname of the current working directory (after all -C options are processed, if any).[1]
```


[1] - [GNU Make manual](https://www.gnu.org/software/make/manual/make.html)

Reply:
Thanks for your advice.